projects/control-service/projects/pipelines_control_service/src/test/java/com/vmware/taurus/authorization/KerberosAuthenticationIT.java

```diff
@@ -0,0 +1,146 @@
+      "datajobs.security.kerberos.keytabFileLocation=target/foo.keytab",
+      "datajobs.security.kerberos.krb5ConfigLocation=target/krb5.conf",
+      "java.security.krb5.conf=target/krb5.conf",
+      "sun.security.krb5.debug=true",
+      "datajobs.authorization.authorized-roles="
+})
+@SpringBootTest(webEnvironment = SpringBootTest.WebEnvironment.RANDOM_PORT, classes = ControlplaneApplication.class)
+@ExtendWith(SpringExtension.class)
+public class KerberosAuthenticationIT {
+
+   /*
+    * Using static variables due to the KDC instance needing to init before
+    * spring startup code from the @SpringBootTest annotation.
+    */
+   private static final String WORK_DIR = "target";
+   private static final File KDC_WORK_DIR = new File(WORK_DIR);
+   private static final File KEYTAB = new File(WORK_DIR + "/foo.keytab");
+   private static final String CLIENT_PRINCIPAL = "client/localhost";
+   private static final String TGT_PRINCIPAL = "HTTP/localhost";
+
+   private static SimpleKdcServer simpleKdcServer;
+
+   @Autowired
+   private JobsRepository jobsRepository;
+
+   private DataJob dataJob;
+
+   @LocalServerPort
+   int randomPort;
+
+   /*
+    * Static block is only reliable way of starting the KDC server before
+    * code invoked by the @SpringBootTest annotation. If you find a way to
+    * refactor this in a JUnit method such as @BeforeAll etc. feel free to.
+    */
+   static {
+      // Initialize the KDC server
+      try {
+         simpleKdcServer = new SimpleKdcServer();
+         simpleKdcServer.setWorkDir(KDC_WORK_DIR);
+         simpleKdcServer.setAllowTcp(true);
+         simpleKdcServer.setAllowUdp(true);
+         simpleKdcServer.setKdcUdpPort(NetworkUtil.getServerPort());
+         // Start the KDC server
+         simpleKdcServer.init();
+         simpleKdcServer.start();
+         simpleKdcServer.createAndExportPrincipals(KEYTAB, CLIENT_PRINCIPAL, TGT_PRINCIPAL);
+      } catch (Exception e) {
+         System.out.println(e);
+      }
+   }
+
+   @BeforeEach
+   public void addTestJob() {
+      // Add a dummy data job so that endpoint call returns 200 instead of 404
+      dataJob = new DataJob();
+      var config = new JobConfig();
+      config.setTeam("test-team");
+      dataJob.setName("test-name");
+      dataJob.setJobConfig(config);
+      jobsRepository.save(dataJob);
+   }
+
+   @Test
+   public void testAuthenticatedCall() throws Exception {
+
+      SpnegoClient spnegoClient = SpnegoClient.loginWithKeyTab(CLIENT_PRINCIPAL, KEYTAB.getPath());
+      URL url = new URL("http://127.0.0.1:" + randomPort + "/data-jobs/for-team/test-team/jobs/test-name");
+      var krb5 = simpleKdcServer.getKrbClient().getKrbConfig();
+      SpnegoContext context = spnegoClient.createContext(new URL("http://" + krb5.getKdcHost() + "/" + krb5.getKdcPort()));
+      HttpURLConnection conn = (HttpURLConnection) url.openConnection();
+      conn.setRequestProperty("Authorization", context.createTokenAsAuthroizationHeader());
```

Comment:
extract this in separate Utility class (KerberosUtility.requestWithKerberosAuth(httpUri, princpak, keytab). 

It's very clear we'd want to reuse this . And generally, this kind of helper code does not belong in the test method itself.  It might be a helper method in the same class. But I think it's better to have a separate utility class since I think there's a very good chance to be reused.